src/setuptools_pyproject_migration/__init__.py

```diff
@@ -91,8 +91,50 @@ def _generate(self) -> Pyproject:
         if dependencies:
             pyproject["project"]["dependencies"] = dependencies
 
+        entrypoints = self._generate_entrypoints()
+
+        # GUI scripts and console scripts go separate in dedicated locations.
+        if "console_scripts" in entrypoints:
+            pyproject["project"]["scripts"] = entrypoints.pop("console_scripts")
+
+        if "gui_scripts" in entrypoints:
+            pyproject["project"]["gui-scripts"] = entrypoints.pop("gui_scripts")
+
+        # Anything left over gets put in entry-points
+        if entrypoints:
+            pyproject["project"]["entry-points"] = entrypoints
+
         return pyproject
 
+    def _parse_entrypoint(self, entrypoint: str) -> Dict[str, str]:
+        """
+        Extract the entry-point and name from the string.
+        """
+        # Format: 'hello-world = timmins:hello_world'
```

Comment:
I think it would be useful to have some tests for this method. Personally I would write a file of unit tests like what I did for `_transform_contributors()` in #45, but if nothing else a few doctests would check the main functionality and help illustrate the method's behavior. If you make it a static method you can do this:

```suggestion
        
        >>> WritePyproject._parse_entrypoint("hello-world = timmins:hello_world")
        {'hello-world': 'timmins:hello_world'}
        >>> WritePyproject._parse_entrypoint("hello-world")
        Traceback (most recent call last):
            ...
        ValueError: Entry point 'hello-world' is not of the form 'name = target'
        """
```

(BTW I am partial to using Monty Python references in the examples but that doesn't really matter of course :stuck_out_tongue:)

Reply:
Yep, I'm in the process of doing that now.  Figured the unit tests would wait until the morning when I had a clear head. :-)